### PR TITLE
update readme

### PR DESCRIPTION
**Pull request by rhagarty**
@aloknsingh - please review.

- "Flow" steps should match the diagram. I left middle step blank for you to fill in.
- I moved your 11 steps down to the "Analyze .." section and consolidated the text (made it 5 steps). We just want to summarize here what the user will see/do in the notebook. All of the details are in the notebook.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,77 +170,38 @@ options to specify exactly what you want shared from your notebook:
...
+The imbalanced dataset is from Purtugese Bank Marketing data, where bank's associate makes call to user to sell financial product i.e CD to bank's client.
 
-2. Data Set Description.
+### 2. Prepare the data
 
- Data set is related to  direct marketing campaigns of a [Portuguese banking institution](http://archive.ics.uci.edu/ml/datasets/Bank+Marketing), where bank's associate makes call to bank's client to sell financial product i.e CD. Data set contains 17 columns and is explained here.
+For this section we will mostly use Python based libraries such as XGBoost, Scikit-learn, Matplotlib, SeaBorn, and Pandas.
 
-3. Statement of Classification Problem.
+### 3. Visual Data Exploration to understand the data using Seaborn and Matplotlib)
```
Comment (stevemar) [inline README.md]: dangling brace ")" at the end here

File: README.md [hunk 2]
```diff
@@ -299,14 +210,13 @@ Awesome job following along! Now go try and take this further or apply it to a d
...
- - Watson Studio: https://datascience.ibm.com/docs/content/analyze-data/creating-notebooks.html.
- - Pandas:http://pandas.pydata.org/
- - Data:http://archive.ics.uci.edu/ml/datasets/Bank+Marketing
- - Scikit Learn:http://scikit-learn.org/stable/
- - XGBoost:https://github.com/dmlc/xgboost
- - Matplotlib:https://matplotlib.org/
- - SeaBorn:https://seaborn.pydata.org
-
+- Watson Studio: https://datascience.ibm.com/docs/content/analyze-data/creating-notebooks.html.
+- Pandas:http://pandas.pydata.org/
```
Comment (stevemar) [inline README.md]: i'd add a space between the colon and the link

**Review by stevemar — APPROVED** [review]
minor comments but i think this is good contribution

**Comment by rhagarty** [comment]
Thanks for review. Issues fixed in latest.

**Comment by aloknsingh** [comment]
@rhagarty @stevemart 
thanks for comments and update.

Here are some thoughts

1) I agree with analyze section consolidation to 5. However, we should add one more last section i.e 
something in the line of following text

"
6. Inference Discussion (Generalization and Prediction)

In many ML training application, there is risk that our model won't generalize enough for unknown data it has not seen. To mitigate it, it is recommended for data scientists to do generalization error testing.
That involves, running cross validated model to predict on held out data, to see it's performance on test data.
It's important that we don't look at held out data or use it in training. Doing so will make our model training bias and generalization error will be huge.
"

**Comment by aloknsingh** [comment]
@rhagarty @stevemart
thanks for comments and update.

Here are some thoughts
Question:
1)"Flow" steps should match the diagram. I left middle step blank for you to fill in.

Answer:
Here is some thoughts for filling in

Actually we can use your consolidated Analyze section heading 
"
1. Log into IBM Watson Studio service.
2. Upload the data as a data asset into Watson Studio.
3. Start a notebook in Watson Studio and input the data asset previously created.
4 .Explore the dataset for get preliminary insights using various visualization and statistics
5. Prepare the data and preprocessing for model training using Scikit-Learn ML pipeline
6. Model Training and evaluation using cross-validation, precision recall metric, ROC metric. 
8. Evaluate their predictive performance on held out data i.e generalization testing
"

**Comment by rhagarty** [comment]
@aloknsingh thanks for feedback. Could I get you to look again at the "Flow" section? The diagram has 8 items that we should describe with the corresponding flow text. We need a short description for:
4 (XGBoost)
5 (Scikit-Learn)
6 (MatplotLib)
7 (pandas). 

We should have descriptions like -
4. Use XGBoost to ....
OR, we change the diagram.

**Comment by aloknsingh** [comment]
@rhagarty thanks again!

Sorry i missed the pointed questions.

Here is one line explanation of each of them


XGBoost: Extreme Gradient Boosting is decision tree based tools for creating ML model. We use XGBoost with optimized parameter to train Model. XGBoost is state of art techniques for creating various classification and regression model. 
Scikit Learn:  We use scikit learn to create ML pipeline for data preparation be be fed into XGBoost.Also many of the classification performance metrics from scikit learn like ROC, precision-recall, auc are used.
Pandas: we use Pandas to read in the csv file into dataframe and for initial data exploration.
Matplotlib:  We use Matplotlib and it's higher level package seaborn for creating various visualization.

**Comment by rhagarty** [comment]
@aloknsingh - ok, I made the changes. Note, we still need to revisit the architecture diagram and flow. We need to change the number to be sequential in operation. For example, Pandas is listed as flow item 7, but is should be list #2 since it involves loading the initial data.

**Comment by rhagarty** [comment]
But lets merge this and work on that later

**Comment by aloknsingh** [comment]
sounds good. I will merge and then re-order pandas to #3 and so on and update architecture diagram
